tests/integrations/test_crossed_integrations.py

```diff
@@ -63,13 +113,29 @@ def get_topic(span) -> str | None:
     def setup_produce(self):
         """
         send request A to weblog : this request will produce a kafka message
-        send request B to python buddy, this request will consume kafka message
+        send request B to library buddy, this request will consume kafka message
         """
-
-        python_buddy = _PythonBuddy()
-
-        self.production_response = weblog.get("/kafka/produce", params={"topic": self.WEBLOG_TO_BUDDY_TOPIC})
-        self.consume_response = python_buddy.get("/kafka/consume", params={"topic": self.WEBLOG_TO_BUDDY_TOPIC})
+        timeout = time.time() + 180
+
+        self.production_response = None
+        self.consume_response = None
+        while time.time() < timeout and (
+            self.production_response is None
+            or self.production_response.status_code != 200
+            or self.production_response.text is None
+        ):
+            self.production_response = weblog.get(
+                "/kafka/produce", params={"topic": self.WEBLOG_TO_BUDDY_TOPIC}, timeout=5
+            )
+
+        while time.time() < timeout and (
+            self.consume_response is None
+            or self.consume_response.status_code != 200
+            or self.consume_response.text is None
+        ):
+            self.consume_response = self.buddy.get(
```

Comment:
lol I think you wrote this test class 🤣 . The basic logic has stayed the same in this PR. But we are producing from the weblog, then consuming from the buddy for `test_produce`. For `test_consume`, we are producing from the buddy and consuming from the weblog. I think this makes sense since during a weblog test such as express4 for nodeJS, if `test_produce` fails, that means that the nodeJS weblog failed to produce, or the language buddy failed to consume.

Reply:
ohhhhhh snap - you're absolutely right!